ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -242,40 +242,103 @@ module Json = struct
             ; ("func_name_suffix", `String (suffix_of_type t))
             ]
 
-  let of_params params =
-    let name_internal name =
-      let name = name |> snake_to_camel |> String.uncapitalize_ascii in
-      match name with "type" -> "typeKey" | "interface" -> "inter" | _ -> name
+  let get_last l = match List.rev l with h :: _ -> Some h | _ -> None
+
+  let group_params msg =
+    let groups params =
+      let params =
+        List.map
+          (fun p -> (published_release_for_param p.param_release.internal, p))
+          params
+      in
+      let uniq_sorted_rels =
+        List.map fst params
+        |> Listext.List.setify
+        |> List.fast_sort compare_versions
+      in
+      List.map
+        (fun rel ->
+          let params =
+            List.filter_map
+              (fun (r, param) ->
+                match compare_versions r rel with
+                | n when n > 0 ->
+                    None
+                | _ ->
+                    Some param
+              )
+              params
+          in
+          (params, rel)
+        )
+        uniq_sorted_rels
     in
-    let of_param param =
-      let suffix_of_type = suffix_of_type param.param_type in
-      let t, _e = string_of_ty_with_enums param.param_type in
-      let name = param.param_name in
+    let only_session_group =
       [
-        ("is_session_id", `Bool (name = "session_id"))
-      ; ("type", `String t)
-      ; ("name", `String name)
-      ; ("name_internal", `String (name_internal name))
-      ; ("doc", `String param.param_doc)
-      ; ("func_name_suffix", `String suffix_of_type)
+        ( [session_id]
+        , published_release_for_param session_id.param_release.internal
+        )
```

Comment:
@minglumlu If it is only session param group, it should be the `msg.msg_release` rather than `session_id.param_release`.